File: app/src/main/java/com/starostinvlad/fan/CurrentSerialDao.java
package com.starostinvlad.fan;

import com.starostinvlad.fan.GsonModels.CurrentSerial;
import com.starostinvlad.fan.GsonModels.Datum;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import io.reactivex.Maybe;
import io.reactivex.Observable;

@Dao
public interface CurrentSerialDao {

    @Query("SELECT * FROM currentSerial")
    Maybe<List<CurrentSerial>> getAll();

    @Query("SELECT * FROM currentSerial WHERE pageId = :pageId")
    Maybe<CurrentSerial> getById(String pageId);

    @Insert
    List<Long> insert(List<CurrentSerial> currentSerials);

    @Insert
    Maybe<Long> insert(CurrentSerial currentSerial);

    @Update
    void update(CurrentSerial currentSerial);

    @Delete
    void delete(CurrentSerial currentSerial);

}

File: app/src/main/java/com/starostinvlad/fan/CurrentSerialDao.kt
package com.starostinvlad.fan

import androidx.room.*
import com.starostinvlad.fan.GsonModels.CurrentSerial
import io.reactivex.Maybe
import kotlin.Throws

@Dao
interface CurrentSerialDao {
    @get:Query("SELECT * FROM currentSerial")
    val all: Maybe<List<CurrentSerial?>?>?

    @Query("SELECT * FROM currentSerial WHERE pageId = :pageId")
    fun getById(pageId: String?): Maybe<CurrentSerial?>

    @Insert
    fun insert(currentSerials: List<CurrentSerial?>?): List<Long?>?

    @Insert
    fun insert(currentSerial: CurrentSerial?): Maybe<Long>

    @Update
    fun update(currentSerial: CurrentSerial?)

    @Delete
    fun delete(currentSerial: CurrentSerial?)
}